tiledb/sm/query/query.h

```diff
@@ -58,8 +58,24 @@
 #include "tiledb/sm/storage_manager/cancellation_source.h"
 #include "tiledb/sm/subarray/subarray.h"
 
+#ifdef HAVE_RUST
+#include "tiledb/oxidize/rust.h"
```

Comment:
How big is `rust.h`? If it adds large amounts of code to the compilation, we should avoid including it in other headers, for example by converting `LogicalExpr` and `Session` to self-contained types, and storing `tdb_unique_ptr`s of them in the class. There is also the concern that if its content is too sensitive to changes in the Rust code's implementation details, it will force recompilations of the C++ code, which will be limited by moving the include to the implementation file.

Reply:
This header is generated by the `cxxbridge` dependency and provides C++ definition which correspond to Rust standard library types, e.g. `rust::Box` and `rust::Vec` and etc.  It isn't generated by compiling any of our Rust crates.